wpiutil/src/main/java/edu/wpi/first/util/sendable2/SendableTable.java

```diff
@@ -0,0 +1,250 @@
+    final StructBuffer<T> buf = StructBuffer.create(struct);
+    publishRawBB(name, struct.getTypeString(), () -> {
+      return buf.write(supplier.get());
+    });
+  }
+
+  default <T> Consumer<T> publishStruct(String name, Struct<T> struct) {
+    addSchema(struct);
+    final StructBuffer<T> buf = StructBuffer.create(struct);
+    final Consumer<ByteBuffer> consumer = publishRawBB(name, struct.getTypeString());
+    return (T value) -> {
+      consumer.accept(buf.write(value));
+    };
+  }
+
+  default <T> void subscribeStruct(String name, Struct<T> struct, Consumer<T> consumer) {
+    addSchema(struct);
+    final StructBuffer<T> buf = StructBuffer.create(struct);
+    subscribeRaw(name, struct.getTypeString(), (data) -> {
+      consumer.accept(buf.read(data));
+    });
+  }
+
+  <T> void setProtobuf(String name, T value, Protobuf<T, ?> proto);
+
+  <T> void publishProtobuf(String name, Protobuf<T, ?> proto, Supplier<T> supplier);
+
+  <T> Consumer<T> publishProtobuf(String name, Protobuf<T, ?> proto);
+
+  <T> void subscribeProtobuf(String name, Protobuf<T, ?> proto, Consumer<T> consumer);
+
+  <T> SendableTable addSendable(String name, T obj, Sendable<T> sendable);
+
+  // SendableTable getSendable(String name);
+
+  SendableTable getChild(String name);
+
+  void setPublishOptions(SendableOption... options);
+
+  void setSubscribeOptions(SendableOption... options);
+
+  /**
+   * Gets the current value of a property (as a JSON string).
+   *
+   * @param name name
+   * @param propName property name
+   * @return JSON string; "null" if the property does not exist.
+   */
+  String getProperty(String name, String propName);
+
+  /**
+   * Sets a property value.
+   *
+   * @param name name
+   * @param propName property name
+   * @param value property value (JSON string)
+   * @throws IllegalArgumentException if properties is not parseable as JSON
+   */
+  void setProperty(String name, String propName, String value);
+
+  /**
+   * Deletes a property. Has no effect if the property does not exist.
+   *
+   * @param name name
+   * @param propName property name
+   */
+  void deleteProperty(String name, String propName);
+
+  /**
+   * Gets all topic properties as a JSON object string. Each key in the object is the property name,
+   * and the corresponding value is the property value.
+   *
+   * @param name name
+   * @return JSON string
+   */
+  String getProperties(String name);
+
+  /**
+   * Updates multiple topic properties. Each key in the passed-in object is the name of the property
+   * to add/update, and the corresponding value is the property value to set for that property. Null
+   * values result in deletion of the corresponding property.
+   *
+   * @param name name
+   * @param properties JSON object string with keys to add/update/delete
+   * @throws IllegalArgumentException if properties is not a JSON object
+   */
+  void setProperties(String name, String properties);
+
+  void erase(String name);
```

Comment:
Should this be "close", if this is about closing resources?
Or "unpublish", if it's not.

The naming of "erase" might be intuitive to C++ programmers, but if we're following STL container verbs then "remove" would be more familiar and intuitive to our Java audience.

Reply:
close() is too overloaded in Java.  I was actually thinking about a rename of this to remove(), so that's probably the right answer.